frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -158,37 +169,55 @@ export const insightLogic = kea<insightLogicType>({
                 setTimeout(() => {
                     if (values && view == values.activeView) {
                         actions.setShowTimeoutMessage(true)
+                        const tags = {
+                            insight: values.activeView,
+                            scene: sceneLogic.values.scene,
+                        }
+                        posthog.capture('insight timeout message shown', tags)
+                        captureInternalMetric({ method: 'incr', metric: 'insight_timeout', value: 1, tags })
                     }
                 }, SHOW_TIMEOUT_MESSAGE_AFTER)
             )
             actions.setIsLoading(true)
         },
-        endQuery: ({ view, lastRefresh, exception }) => {
-            clearTimeout(values.timeout || undefined)
+        endQuery: ({ queryId, view, lastRefresh, exception }) => {
+            if (values.timeout) {
+                clearTimeout(values.timeout)
+            }
             if (view === values.activeView) {
                 actions.setShowTimeoutMessage(values.maybeShowTimeoutMessage)
                 actions.setShowErrorMessage(values.maybeShowErrorMessage)
                 actions.setLastRefresh(lastRefresh || null)
                 actions.setIsLoading(false)
-                if (values.maybeShowTimeoutMessage) {
-                    posthog.capture('insight timeout message shown', { insight: values.activeView, ...exception })
```

Comment:
Is this not an error case anymore? Or is it that this is actually an error and not a timeout (per your comment elsewhere). 

If the latter, we have this `insight error message shown` event – could we use that here instead so we can be consistent with old data?

Reply:
Not sure I understand, so let me give some context.

Previously this metric was wrong/misleading:
1. It was reported when the query did not time out, but got another exception
2. If user ran query, saw the timeout message and left the page, this event was never captured

After this PR change we capture "insight timeout message shown" when the message is shown - not when it finishes or when an unrelated error occurs.

Only thing worse than a missing metric is a misleading one.